spec/controllers/queue_items_controller_spec.rb

```diff
@@ -102,9 +114,75 @@
+      it_behaves_like "requires sign in" do
+        let(:action) { delete :destroy, id: queue_item.id }
+      end
+    end
+  end
+
+  describe "POST update_queue" do
+    context "with valid inputs" do
+      let(:laura) {Fabricate(:user)}
+      let(:video) {Fabricate(:video)}
+      let(:queue_item1) {Fabricate(:queue_item, user: laura, position: 1, video: video)}
+      let(:queue_item2) {Fabricate(:queue_item, user: laura, position: 2, video: video)}
+      before do
+        set_current_user(laura)
+      end
+      it "redirects to the my queue page" do
+        post :update_queue, queue_items: [{id: queue_item1.id, position: 2}, {id: queue_item2.id, position: 1}]
+        expect(response).to redirect_to my_queue_path
+      end
+      it "reorders the queue items" do
+        post :update_queue, queue_items: [{id: queue_item1.id, position: 2}, {id: queue_item2.id, position: 1}]
+        expect(laura.queue_items).to eq([queue_item2, queue_item1])
+      end
+      it "normalizes the position numbers" do
+        post :update_queue, queue_items: [{id: queue_item1.id, position: 3}, {id: queue_item2.id, position: 2}]
+        expect(laura.queue_items.map(&:position)).to eq([1,2 ])
+      end
+    end
+
+    context "with invalid inputs" do
+      let(:laura) {Fabricate(:user)}
+      let(:video) {Fabricate(:video)}
+      let(:queue_item1) {Fabricate(:queue_item, user: laura, position: 1, video: video)}
+      let(:queue_item2) {Fabricate(:queue_item, user: laura, position: 2, video: video)}
+      before do
+        set_current_user(laura)
+      end
+      it "redirects to the my queue page" do
+        post :update_queue, queue_items: [{id: queue_item1.id, position: 3.5}, {id: queue_item2.id, position: 2}]
+        expect(response).to redirect_to my_queue_path
+      end
+      it "sets the flash notice" do
+        post :update_queue, queue_items: [{id: queue_item1.id, position: 3.5}, {id: queue_item2.id, position: 2}]
+        expect(flash[:error]).to be_present
+      end
+      it "does not change the queue items" do
+        post :update_queue, queue_items: [{id: queue_item1.id, position: 3}, {id: queue_item2.id, position: 2.1}]
+        expect(queue_item1.reload.position).to eq(1)
+      end
+    end
+    
+    context "with unauthenticated users" do
+      it_behaves_like "requires sign in" do
+        let(:action) { post :update_queue, queue_items: [{id: 10, position: 3}, {id: 11, position: 2}] }
+      end
+    end
+
+    context "with queue items that do not belong to the current user" do
+      let(:laura) {Fabricate(:user)}
+      let(:ryan) {Fabricate(:user)}
+      let(:video) {Fabricate(:video)}
+      let(:queue_item1) {Fabricate(:queue_item, user: laura, position: 1, video: video)}
+      let(:queue_item2) {Fabricate(:queue_item, user: ryan, position: 2, video: video)}
+      before do
+        set_current_user(laura)
+      end
+      it "should not update the queue items" do
```

Comment:
Do not use "should" in example message - it does not bring any value added to the spec, `it "does not update queue items" is what spec is expecting.